### PR TITLE
Restrict timeline collection to a subset of demos

### PR DESCRIPTION
**Pull request by cbracken**
Also refactors demos list into a class with `synchronized` and
`profiled` properties.

When run with --trace-startup, as we do in this test, the VM stores
trace events in an endless buffer instead of a ring buffer. To avoid
out-of-memory conditions, we restrict timeline collection to a subset.

### CODE REVIEW

File: examples/flutter_gallery/test_driver/transitions_perf_test.dart [hunk 1]
```diff
@@ -11,57 +11,72 @@ import 'package:flutter_driver/flutter_driver.dart';
...
+
+  // True if timeline data should be collected for this test.
+  //
+  // Warning: The number of tests executed with timeline collection enabled
+  // significantly impacts heap size of the running app. When run with
+  // --trace-startup, as we do in this test, the VM stores trace events in an
+  // endless buffer instead of a ring buffer.
+  final bool profiled;
+
+  const Demo(this.title, {bool this.synchronized = true, bool this.profiled = false});
```
Comment (Hixie) [inline examples/flutter_gallery/test_driver/transitions_perf_test.dart]: nit: constructor first

Comment (cbracken) [inline examples/flutter_gallery/test_driver/transitions_perf_test.dart]: Done.

**Comment by Hixie** [comment]
LGTM

Please wait for current test failures to be resolved before checking this in. Thanks.

**Comment by cbracken** [comment]
Fixes #9034. If we can come up with a long-term fix that avoids buffering so much timeline data, re-enabling timeline data collection for all tests seems reasonable.

**Comment by eseidelGoogle** [comment]
FYI @yjbanov @chinmaygarde @johnmccutchan

**Review by yjbanov — APPROVED** [review]

**Pull request merged**